src/fragments/lib/push-notifications/common/request-permissions.mdx

```diff
@@ -0,0 +1,61 @@
+Depending on your users' platform and operating system version, it is likely that you will need to request their permission to display push notifications.
+
+To learn more about the platform-specific guidances for requesting permissions, you can visit the respective documentations for [iOS](https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications) and [Android](https://developer.android.com/develop/ui/views/notifications/notification-permission). To best aid you in giving your users a good permission experience with platform idiomatic flows, Amplify provides the functionality below.
+
+## Get permission status
+
+The first step to request permissions from your user is to understand the current status of permissions. Your app may behave differently in response to these possible statuses below.
+
+- **Should Request** - No permissions have been requested yet. It is idiomatic at this time to simply request for permissions from the user.
+- **Should Explain Then Request** - It is recommended at this time to provide some context or rationale to the user explaining why you are requesting permission to display notifications to them before then requesting for permissions.
```

Comment:
```suggestion
- **Should Explain Then Request** - It is recommended at this time to provide some context or rationale to the user explaining why you want to send them push notifications before requesting for permissions.
```

Reply:
Agreed with this as well ^

Reply:
@cshfang - wanna accept this so that I can approve the PR?